### PR TITLE
Weapons 2

### PR DESCRIPTION
**Pull request by TMTIME**
your move fonsi

:cl: TMTIME
add: Player ship evasion is now stored in map configs
add: Fixed number of weapon hardpoints per map
add: Phase cannon chips now for sale at stores.
add: Lenses in weapon hardpoints can now be made from glass at a 3X miss rate increase.
add: Ion damage. Tldr turn off a component for a period of time
tweak: Evasion tweaked
/:cl:

wew what did I even do here?

Player evasion is now stored in map configs. So now the FOBshuttle has 45% base evasion

Maps now have a fixed number of weapons. Smaller ships like the Butterfly only have three, the majority have 4 and the Scarab has 6. ~~Fuck knows how many the FOBshuttle has I didn't bother counting~~

A weapon that is blown up reverts to a hardpoint.

Weapons accuracy is also a % mod.

Phase cannon chips can now be ordered for 4k. This absolutely needs balancing.

Lenses in weapon hardpoints can now take glass as well as diamond. Glass lenses incurs a 3* miss rate for the weapon

I O N D A M A G E

Ions now only do huge shield damage and turn off components for ~10 seconds
t a c t i c a l g a m e p l a y

~~also nerfed the fun ship due to weapons revamp~~

### CODE REVIEW

**Comment by Kmc2000** [comment]
woah

**Comment by NIXC** [comment]
If we are giving out free evasion boosts can the Butterfly have one as well? Its pretty small, and doesn't like getting hit, it would make sense it was more evasive.

**Comment by TMTIME** [comment]
Remember its not purely about size, its about number of engines as well.

Ideally we'd have a throttle where the helms can increase speed (and therfor evasion) at a heavy drain to power

File: code/controllers/subsystem/ships.dm [hunk 1]
```diff
@@ -18,7 +18,7 @@ SUBSYSTEM_DEF(ship)
 	var/error_sound = 'sound/machines/buzz-sigh.ogg'
 	var/notice_sound = 'sound/machines/twobeep.ogg'
 
-	var/player_evasion_chance = 25 //evasion chance for the player ship
+	var/player_evasion_chance = 15 //evasion chance for the player ship
```
Comment (Qustinnus) [inline code/controllers/subsystem/ships.dm]: add that this is overriden in the config

File: code/controllers/subsystem/ships.dm [hunk 2]
```diff
@@ -132,10 +132,17 @@ SUBSYSTEM_DEF(ship)
 			C = pick(S.ship_components)
 			if(C.active) C = null
 		if(C)
+			C.health += 1 //Only have time for basic functionality.
```
Comment (Qustinnus) [inline code/controllers/subsystem/ships.dm]: lmao this wasn't there? goddamnit viv

File: code/controllers/subsystem/ships.dm [hunk 3]
```diff
@@ -132,10 +132,17 @@ SUBSYSTEM_DEF(ship)
...
 			C.active = 1 //fix that shit
 
 			broadcast_message("<span class=notice>[faction2prefix(S)] ship ([S.name]) has repaired [C.name] at ([C.x_loc],[C.y_loc]).</span>",notice_sound,S)
 
+/datum/controller/subsystem/ship/proc/restore_component_power(var/datum/ship_component/C)
+	if(C) //haha runtimes am I right?
+		if(C.ship) //TODO, fix the need for these checks
```
Comment (Qustinnus) [inline code/controllers/subsystem/ships.dm]: todo: kill the fucking ship subsystem. I did a large rework of it once but never finished it so I can try add that once we rebase.

also theres comment syntax for properly showing todo tasks. I think its TODO:

File: code/controllers/subsystem/ships.dm [hunk 4]
```diff
@@ -240,7 +247,15 @@ SUBSYSTEM_DEF(ship)
 				if(istype(S)) // fix for runtime (ship might have ceased to exist during the spawn)
 					broadcast_message("<span class=notice>Shot hit [faction2prefix(S)] shields. [faction2prefix(S)] ship shields at [S.shield_strength / initial(S.shield_strength) * 100]%!</span>",notice_sound,S)
 		return
+	if(attack_data.emp_attack && C.online) //If we do EMP damage, turn the component off
```
Comment (Qustinnus) [inline code/controllers/subsystem/ships.dm]: we should make emp_attack a flag for attack_data. not necesary for this PR but is something to remember

Comment (Qustinnus) [reply code/controllers/subsystem/ships.dm]: or is it a numeric value? cba to check :")

File: code/modules/ftl/machinery/phase_cannon.dm [hunk 5]
```diff
@@ -176,43 +181,54 @@
...
-			if(istype(W, /obj/item/stack/sheet/mineral/diamond)) //Contruct
-				var/obj/item/stack/sheet/mineral/diamond/D = W
-				if(D.use(5))
-					to_chat(user, "<span class='notice'>You install \the [D] as focusing lenses for \the [src].</span>")
-					state = CONSTRUCTION_STATE4
-				else
-					to_chat(user, "<span class='warning'>You require at least 5 [D] to install lenses for \the [src]!</span>")
+			if(istype(W, /obj/item/stack/sheet)) //Contruct
+				var/obj/item/stack/sheet/S = W
+				if(istype(S, /obj/item/stack/sheet/mineral/diamond)) //|| istype(S, /obj/item/stack/sheet/glass))
```
Comment (Qustinnus) [inline code/modules/ftl/machinery/phase_cannon.dm]: is there a reason for this comment

File: code/modules/ftl/machinery/phase_cannon.dm [hunk 6]
```diff
@@ -176,43 +181,54 @@
...
+				state = CONSTRUCTION_STATE4
+
 
 		if(CONSTRUCTION_STATE4) //remove lenses or add chip
 			if(istype(W, /obj/item/weapon/crowbar)) //Decon
-				to_chat(user, "<span class='notice'>You pry the focusing lenses out of \the [src].</span>")
-				var/obj/item/stack/sheet/mineral/diamond/D = new (loc, 5)
-				D.add_fingerprint(user)
+				to_chat(user, "<span class='notice'>You pry \the [lens] focusing lenses out of \the [src].</span>")
+				lens.loc = src.loc
```
Comment (Qustinnus) [inline code/modules/ftl/machinery/phase_cannon.dm]: im pretty sure theres a proc like put_in_hands(). i know i didnt use that for my pr either but its something to look into, isnt really necesary but might be nice.

File: code/modules/ftl/machinery/phase_cannon.dm (hunk 5 above)
Comment (TMTIME) [reply code/modules/ftl/machinery/phase_cannon.dm]: Originally I used to use that as one large if statement, but it became easier to split them, so done and forgot about.

File: code/controllers/subsystem/ships.dm (hunk 4 above)
Comment (TMTIME) [reply code/controllers/subsystem/ships.dm]: ~~disregard the statement or the flag bit?~~

File: code/modules/ftl/machinery/phase_cannon.dm (hunk 6 above)
Comment (TMTIME) [reply code/modules/ftl/machinery/phase_cannon.dm]: I'm gonna say we don't need it? Machine frames currently just dump components on the floor, so the current method just emulates that.

File: code/controllers/subsystem/ships.dm (hunk 4 above)
Comment (Qustinnus) [reply code/controllers/subsystem/ships.dm]: this entire comment

**Pull request merged**